src/SALib/sample/morris/morris.py

```diff
@@ -303,42 +300,27 @@ def _compute_optimised_trajectories(problem: dict, input_sample: int, N: int,
         The number of optimal trajectories
     local_optimization : bool, default=False
         If true, uses local optimisation heuristic
-    '''
-    if _has_gurobi is False \
-            and local_optimization is False \
-            and k_choices > 10:
+    """
+    if local_optimization is False and k_choices > 10:
         msg = "Running optimal trajectories greater than values of 10 \
                 will take a long time."
         raise ValueError(msg)
 
-    num_params = problem['num_vars']
-
     if np.any((input_sample < 0) | (input_sample > 1)):
         raise ValueError("Input sample must be scaled between 0 and 1")
 
-    if _has_gurobi and local_optimization is False:
-        # Use global optimization method
-        strategy = GlobalOptimisation()
-    elif local_optimization:
-        # Use local method
-        strategy = LocalOptimisation()
-    else:
-        # Use brute force approach
-        strategy = BruteForce()
-
-    if problem.get('groups'):
-        num_groups = len(set(problem['groups']))
-    else:
-        num_groups = None
+    num_groups = len(set(problem['groups']))
```

Comment:
Groups are now always present so we don't need this check?

Reply:
That actually came with my last contribution. My commit description wasn't really accurate, because I wanted to be short. But, what happens in the code now is that, if the user doesn't define groups, the function `_define_problem_with_groups` makes the `groups` key equal to the variables names. In other words, the number of groups will be equal to the number of variables, which is equivalent to no groups. This eliminates some `if` statements, including this one, and also some extra code.